randovania/game_description/editor.py

```diff
@@ -65,37 +73,68 @@ def sub(n: Node):
         new_identifier = dataclasses.replace(old_identifier, area_name=new_name)
 
+        self.replace_references_to_area_identifier(
+            old_identifier,
+            new_identifier,
+        )
+
+        new_area = dataclasses.replace(current_area, name=new_name)
+        current_world.areas[current_world.areas.index(current_area)] = new_area
+
+        self.game.world_list.invalidate_node_cache()
+
+    def replace_references_to_area_identifier(self, old_identifier: AreaIdentifier, new_identifier: AreaIdentifier):
+        if old_identifier == new_identifier:
+            return
+
         for world in self.game.world_list.worlds:
             for area in world.areas:
                 for i in range(len(area.nodes)):
                     node = area.nodes[i]
-
                     new_node = None
+
                     if isinstance(node, TeleporterNode):
                         if node.default_connection == old_identifier:
-                            new_node = dataclasses.replace(node,
-                                                           name=node.name.replace(old_name, new_name),
-                                                           default_connection=new_identifier)
+                            new_node = dataclasses.replace(
+                                node,
+                                name=node.name.replace(old_identifier.area_name, new_identifier.area_name),
+                                default_connection=new_identifier,
+                            )
 
                     elif isinstance(node, DockNode):
-                        if world == current_world and node.default_connection.area_name == old_name:
+                        if node.default_connection.area_identifier == old_identifier:
                             new_node = dataclasses.replace(
                                 node,
-                                name=node.name.replace(old_name, new_name),
-                                default_connection=DockConnection(
-                                    area_name=new_name,
-                                    dock_index=node.default_connection.dock_index,
+                                name=node.name.replace(old_identifier.area_name, new_identifier.area_name),
+                                default_connection=dataclasses.replace(
+                                    node.default_connection,
+                                    area_identifier=new_identifier,
                                 ),
                             )
 
                     if new_node is not None:
                         self.replace_node(area, node, new_node)
 
-        new_area = dataclasses.replace(current_area, name=new_name)
-        current_world.areas[current_world.areas.index(current_area)] = new_area
+    def replace_references_to_node_identifier(self, old_identifier: NodeIdentifier, new_identifier: NodeIdentifier):
+        if old_identifier == new_identifier:
+            return
```

Comment:
you've got some redundant checks for the same condition before calling this function, could just remove those and let the function handle it